libraries/AP_IQ_Motor/AP_IQ_Motor.cpp

```diff
@@ -0,0 +1,263 @@
+ * The AP_IQ_Motor object provides functions necessary to communicate with Vertiq modules.
+*/
+
+
+#include "AP_IQ_Motor.h"
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <SRV_Channel/SRV_Channel.h>
+
+// #include <AP_Logger/AP_Logger.h>
+
+#include <GCS_MAVLink/GCS.h>
+
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_IQ_Motor::var_info[] = {
+
+    // @Param: CTRL_LEN
+    // @DisplayName: Number of Motors
+    // @Description: Sets how many motors will be sent throttle commands starting from 0 and being contiguous
+    // @Range: 0 16
+    // @User: Advanced
+    AP_GROUPINFO("LEN", 1, AP_IQ_Motor, _broadcast_length, 0),
+
+    // @Param: TELEM_MASK
+    // @DisplayName: Motor Telemetry Bitmask
+    // @Description: Sets which motors are sending data back to the flight controller
+    // @Range: 0 65535
+    // @User: Advanced
+    AP_GROUPINFO("TELEM", 2, AP_IQ_Motor, _telemetry_bitmask, 0),
+
+    AP_GROUPEND
+};
+
+AP_IQ_Motor::AP_IQ_Motor(void)
+{
+    AP_Param::setup_object_defaults(this, var_info);
+}
+
+void AP_IQ_Motor::init(void)
+{
+    //Grab access to the global serial manager if it exists
+    AP_SerialManager *serial_manager = AP_SerialManager::get_singleton();
+    if (!serial_manager) {
+        return;
+    }
+
+    //Try to find an instance of a serial peripheral configured to use the IQ protocol
+    _iq_uart = serial_manager->find_serial(AP_SerialManager::SerialProtocol_IQ, 0);
+
+    // check if the uart exists
+    if (_iq_uart) {
+        //Grab the number of IFCI values the user wants to use
+        const int8_t length = _broadcast_length.get();
+
+        //Limit the user's length parameter to only the number of available channels
+        _total_channels = NUM_SERVO_CHANNELS > length ? length : NUM_SERVO_CHANNELS;
+
+        //Find the position of the first and last motor that we will request telemetry from
+        const uint8_t bitmask = _telemetry_bitmask.get();
+        bool found_first = false;
+        for (uint8_t motor = 0; motor < NUM_SERVO_CHANNELS; ++motor) {
+            if (bitmask & 1 << motor) {
+                if (found_first) {
+                    _last_telem_motor = motor;
+                } else {
+                    _first_telem_motor = motor;
+                    found_first = true;
+                }
+            }
+        }
+    }
+}
+
+uint8_t AP_IQ_Motor::add_client(ClientAbstract *new_client)
+{
+    //Make sure that there is enough space to add a new client
+    if (_client_size < _client_limit) { // TODO check if this client type already exists and don't add if it does. Instead return the pointer?
+        _clients[_client_size] = new_client;
+        _client_size++;
+        return 1;
+    }
+
+    return 0;
+}
+
+uint8_t AP_IQ_Motor::find_client(ClientAbstract **set_client, uint8_t client_type, uint8_t obj_idn)
+{
+    //Go and try to find the client specified in the function call
+    for (int client = 0; client < _client_size; ++client) {
```

Comment:
Some style stuff would be nice to fix if possible.  so for example we normally use "int16_t" instead of "int" because it can be different on different platforms.

AP variable name is normally done with underscores instead of CamelCase (or Pascal case).